source/agora/consensus/Fee.d

```diff
@@ -401,6 +401,29 @@ public class FeeManager
         } ();
     }
 
+    /***************************************************************************
+
+        Calculate the total fees of a transaction (including the payload fee)
+
+        Params:
+            tx = transaction for which we want to calculate the fee
+            peekUTXO = UTXO finder
+
+        Returns: the total fee of the transaction (including the payload fee)
+
+    ***************************************************************************/
+
+    public Amount getTXFee (in Transaction tx, scope UTXOFinder peekUTXO) nothrow @safe
+    {
+        Amount tot_fee;
+        Amount tot_data_fee;
+
+        getTXSetFees([tx], peekUTXO, tot_fee, tot_data_fee);
+        tot_fee.add(tot_data_fee);
```

Comment:
Bypasses the overflow check

Reply:
That was actually intentional. This method can be called with a partially validated transaction, and the fee might very well overflow. I can certainly not `assert` here, as this transaction can come from an attacker, and when that happens all debug build of agora intstance would crash. It should be the responsibility of the caller to check for validity and decide what to do. I also would not like to fully validate the input transaction before it is passed to this function, as I would like to fail fast, if the fee threshold is not respected.

I will modify the documentation and mention that the result should be checked. Alternatively I could change the interface and return a bool.